### ISSUE TITLE
auto suggestion don't work properly for houdini module's

### ISSUE THREAD

**Issue by beka202** [microsoft/pylance-release#1799]
Hello,
I use  visual studio code for  Houdini and call hou module. 
Your software suggests me only one generation methods of hou module and this is a serious problem.
for example: if i use hou.node() method, it returns the object, which has again node() and many methods, but your suggestion is not working anymore. 
it's a big bug and needs to fix.
This works perfectly in houdini's internal python shell, so it is doable i think.
Thanks for your time!

**Comment by jakebailey** [comment]
Please provide a code example and some instructions on how we can reproduce; you didn't seem to fill out our issue template. Some of the instructions are mirrored here: https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue

> This works perfectly in houdini's internal python shell, so it is doable i think.

Be aware that this isn't an indicator of anything; when you run the code, you have all of the info to inspect it. Pylance does not execute code to analyze it, it's a static analyzer and there are behaviors that are "legal" at runtime that cannot be handled by only looking at the code.

**Comment by beka202** [comment]
Anyway, thank you, it seems that a dynamic analyzer is needed to reproduce it.
But, I can tell you an example shortly:import hou
hou.node("/obj").node("geo").node("box")  refer to box node which is in the geo node. node() is a method, but after calling this method for the first time (green color), the next one (red color) cannot be recognized
All the best,Beka

    On Friday, September 10, 2021, 09:44:58 PM GMT+4, Jake Bailey ***@***.***> wrote:  
 
 


Please provide a code example and some instructions on how we can reproduce; you didn't seem to fill out our issue template. Some of the instructions are mirrored here: https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue


This works perfectly in houdini's internal python shell, so it is doable i think.


Be aware that this isn't an indicator of anything; when you run the code, you have all of the info to inspect it. Pylance does not execute code to analyze it, it's a static analyzer and there are behaviors that are "legal" at runtime that cannot be handled by only looking at the code.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by jakebailey** [comment]
Houdini appears to be a proprietary paid 3D modelling software; we can't really test this ourselves.

It would be more helpful to provide logs and screenshots of the behavior you're seeing. https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue

What happens when you go-to-def on the first node call?

**Comment by beka202** [comment]
Basically a first node calling returns the same class object, which has the same methods, but your system propably don't recognize class objects after calling a method and
respectively seeing anything.There is the thing.If i firstly call the class, then  the second call is recognized.
Sohou.class.method().method2()
green collor means to work

    On Friday, September 10, 2021, 11:36:08 PM GMT+4, Jake Bailey ***@***.***> wrote:  
 
 


Houdini appears to be a proprietary paid 3D modelling software; we can't really test this ourselves.

It would be more helpful to provide logs and screenshots of the behavior you're seeing. https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue

What happens when you go-to-def on the first node call?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by beka202** [comment]
Sorry i had to say that respectively seeing nothing.

    On Saturday, September 11, 2021, 12:26:02 AM GMT+4, Beka Baratashvili ***@***.***> wrote:  
 
  
Basically a first node calling returns the same class object, which has the same methods, but your system propably don't recognize class objects after calling a method and
respectively seeing anything.There is the thing.If i firstly call the class, then  the second call is recognized.
Sohou.class.method().method2()
green collor means to work

    On Friday, September 10, 2021, 11:36:08 PM GMT+4, Jake Bailey ***@***.***> wrote:  
 
 


Houdini appears to be a proprietary paid 3D modelling software; we can't really test this ourselves.

It would be more helpful to provide logs and screenshots of the behavior you're seeing. https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue

What happens when you go-to-def on the first node call?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by jakebailey** [comment]
The only thing I can think is that the return type for `node` is wrong, but this is speculation. We would need the code, screenshots, logs, anything; descriptions aren't enough to look into this.

**Comment by beka202** [comment]
Yeah and there is hundred and hundreds methods and not only node() which has the same problem.In summary i think that you were right. Houdini's self python shell is dynamical -  when i write each line of code, it Immediately gives me the result.So it's imposible to fix it.I am sorry for taking your time and thank you again


    On Saturday, September 11, 2021, 12:34:15 AM GMT+4, Jake Bailey ***@***.***> wrote:  
 
 


The only thing I can think is that the return type for node is wrong, but this is speculation. We would need the code, screenshots, logs, anything; descriptions aren't enough to look into this.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by jakebailey** [comment]
>  Houdini's self python shell is dynamical -  when i write each line of code, it Immediately gives me the result.So it's imposible to fix it.

This isn't exactly what I'm saying. Where are you editing your code? In .py files in VS Code with Pylance, or are you writing code directly in something else where it's interpreting the code?

**Comment by beka202** [comment]
I am typing code  In .py files in VS Code with Pylance, but Houdini has also a different typing type - dynamical dircet python shell in which auto suggestion works perfectly on every step
    On Monday, September 13, 2021, 09:19:36 PM GMT+4, Jake Bailey ***@***.***> wrote:  
 
 



Houdini's self python shell is dynamical - when i write each line of code, it Immediately gives me the result.So it's imposible to fix it.


This isn't exactly what I'm saying. Where are you editing your code? In .py files in VS Code with Pylance, or are you writing code directly in something else where it's interpreting the code?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by jakebailey** [comment]
Right, I would ignore the shell; that is a running Python interpreter which of course will have perfect fidelity, similar to the REPL or a Python notebook.

I think the key thing here is knowing where these houdini modules are and if they are resolvable. If this is embedded, they may be in some folder, or maybe even unavailable, but that's where you'd be having to look and see.

**Comment by beka202** [comment]
Yeah, sorry i can not provide you more information about that, but I hope that someday someone can help you solve this issue better than me.

   On Monday, September 13, 2021, 09:48:29 PM GMT+4, Jake Bailey ***@***.***> wrote:  
 
 


Right, I would ignore the shell; that is a running Python interpreter which of course will have perfect fidelity, similar to the REPL or a Python notebook.

I think the key thing here is knowing where these houdini modules are and if they are resolvable. If this is embedded, they may be in some folder, or maybe even unavailable, but that's where you'd be having to look and see.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
